### ISSUE TITLE
devise doesn't seem to record user when submitting new item

### ISSUE THREAD

**Issue by mikeducey** [mikeducey/sandbox_for_devise#1]
I am following a guide that is a simple add a bookmark link to a list of bookmarks for users.  But, when I try and add a new item, or edit an existing bookmark, I get an error.  The error is attached below.
![screen shot 2016-08-11 at 1 39 26 pm](https://cloud.githubusercontent.com/assets/19538441/17600461/3dc58cd2-5fc9-11e6-97fb-05a182d94823.png)

I think that there is an issue with the form, for it seems that when I click submit, it does not register that the user exists, or that the user cannot be blank.  (Not sure if they are the same thing!)

I don't see anything on the form submission (which I copied from the tutorial) having anything about transferring, or even requiring user data in order to submit info.  I'm wondering if there is an issue with devise not recognizing that there is a user that is logged in? 

This issue might not be specific to this form, but to a lack of understanding about devise in general.  Any help is appreciated...

**Comment by halfghaninne** [comment]
I'm not sure if there is a problem with Devise, but for now let's _expect_ that User log in is working the way it is supposed to (it might _not_ be but we can address that after a little preliminary debugging).

When I read the error here, it suggests to me that something with the Bookmark object's creation is dependent on a User object. Sure enough, looking in **db/schema.rb**, I see that the `bookmarks` table has a column for `user_id`. And in the Bookmark model, I see that a Bookmark `belongs_to :user`.

Are you passing in information about the current user (the User is ostensibly logged in with Devise) when creating a new bookmark? Taking a look at the **_form** view and `create` controller action for bookmarks doesn't seem to indicate so.

**Comment by mikeducey** [comment]
I can get it to work when removing that authentication from the bookmark class, but that won't be good, as it will remove validation for it on all pages, correct?

I commented out validates :user_id, presence: true and it works. But is that a solution that will be a long term one?  Is the validation only going to matter during that form submission?

**Comment by halfghaninne** [comment]
To be extra clear (and for the sake of recording the conversation) removing the validation for `:user_id` is **definitely not ideal**. Your relationship between a user and their bookmarks depends on the presence of a `user_id` for each Bookmark object.

**Comment by mikeducey** [comment]
This is helpful, but I just am not sure how I can pass that user_id info on when clicking submit.  I can see how it's done when there is an area where there is info to be submitted.  My weakness in form submissions is showing here.  I don't exactly know the format in rails for passing hidden variables (or if, even, that the user_id will be seen when I pass it on!  Any idea of how to learn this, or where to go for a tutorial?

**Comment by halfghaninne** [comment]
There are a couple of options that I can think of:
1. Using hidden form input to pass in current_user.id as a part of the form params
   - Here, I would Google for "hidden input rails 4 form" and see what turns up
2. Assigning current_user.id as the bookmark object's `user_id` in the controller action before you save it.
   - Here, I might experiment with `binding.pry` [here](https://github.com/mikeducey/sandbox_for_devise/blob/master/app/controllers/bookmarks_controller.rb#L31) to think about how this might be accomplished.  

I prefer the first, but I think both are valid.

**Comment by mikeducey** [comment]
Okay, I swear I've made progress, and binding.pry did actually help once I figured out where to put it (I actually used it to help know that I am passing on the user_id value, by putting it on another page, so I'm happy about that.  BUT, I still am unsure of my hidden values being passed, since I am experiencing the error.

Here are my thoughts:  the user's information is stored under the value `current_user`.  So, in my form, I tried to pass it using  `<%= f.hidden_field :user_id, value: current_user %>`.  I still get the error, and binding.pry tells me that `user_id` is  `nil` value.   So, I don't know if I'm trying to pass that information correctly. I am wondering if "value" is what I should use, as the form as it exists does not  pass the values for `:title` and `:url` using value.  But since the value for those variables is what is being inputted, it could be different (and thus "value" still be something I should use).
